extensions/dstore.js

```diff
@@ -5,15 +5,20 @@ define(['../util/lang', '../Variable'], function(lang, Variable){
 	{
 		gotValue: function(value, context) {
 			value = Variable.prototype.gotValue.call(this, value, context)
-			if (value && value.on){
+			if (value && value.on && !(this.seen || (this.seen = new Set())).has(value)){
```

Comment:
perhaps we can just keep the last value (not `returnedVariable`) instead of `Set`, if handlers are reliably being deregistered